pkg/scaling/resolver/hashicorpvault_handler.go

```diff
@@ -116,23 +124,27 @@ func (vh *HashicorpVaultHandler) token(client *vaultapi.Client) (string, error)
 			vh.vault.Credential = &defaultCred
 		}
 
-		if len(vh.vault.Credential.ServiceAccount) == 0 {
-			return token, errors.New("k8s SA file not in config")
+		if vh.vault.Credential.ServiceAccountName == "" && len(vh.vault.Credential.ServiceAccount) == 0 {
```

Comment:
nit: for consistency reasons, can it check for empty string the same way please?

I'm fine with either, but slight preference for `str == ""` check
```suggestion
		if vh.vault.Credential.ServiceAccountName == "" && vh.vault.Credential.ServiceAccount == "" {
```